src/Microsoft.AspNetCore.Razor.Language/CodeGeneration/DesignTimeTagHelperWriter.cs

```diff
@@ -32,26 +38,23 @@ public override void WriteAddTagHelperHtmlAttribute(CSharpRenderingContext conte
             context.RenderChildren(node);
         }
 
-        public override void WriteCreateTagHelper(CSharpRenderingContext context, CreateTagHelperIRNode node)
-        {
-            var tagHelperVariableName = GetTagHelperVariableName(node.TagHelperTypeName);
-
-            context.Writer
-                .WriteStartAssignment(tagHelperVariableName)
-                .WriteStartMethodInvocation(
-                    CreateTagHelperMethodName,
-                    "global::" + node.TagHelperTypeName)
-                .WriteEndMethodInvocation();
-        }
-
-        public override void WriteExecuteTagHelpers(CSharpRenderingContext context, ExecuteTagHelpersIRNode node)
-        {
-            // Do nothing
-        }
-
         public override void WriteInitializeTagHelperStructure(CSharpRenderingContext context, InitializeTagHelperStructureIRNode node)
         {
             context.RenderChildren(node);
+
+            // CreateTagHelper
+            foreach (var descriptor in node.TagHelperBinding.Descriptors)
```

Comment:
Having TagHelperIRNode be the source of truth. Aka, what the binding was, what the tag name, tag mode etc. is makes it easy to understand. With how it is right now: Why would a `TagHelperBodyIRNode` need to have knowledge of the binding?